src/box.jl

```diff
@@ -2,17 +2,19 @@ export Box
 
 type Box{N,D,L} <: Shape{N}
     c::SVector{N,Float64} # box center
-    p::SMatrix{N,N,Float64,L} # projection matrix to box coordinates
     r::SVector{N,Float64}   # "radius" (semi-axis) in each direction
+    p::SMatrix{N,N,Float64,L} # projection matrix to box coordinates
     data::D             # auxiliary data
+    (::Type{Box{N,D,L}}){N,D,L}(c,r,p,data) = new{N,D,L}(c,r,p,data)
```

Comment:
By the way, the complicated form of the inner constructor is to support both Julia 0.5 and 0.6, as mentioned in https://github.com/stevengj/GeometryPrimitives.jl/pull/2#issue-234981153.  I will replace this to the straightforward form when dropping the support for 0.5 in the next PR.